lib/utils/TreePublisher.ts

```diff
@@ -0,0 +1,20 @@
+import {BlueshellState} from '../nodes/BlueshellState';
+
+// @@@ probably a better way to do this than a global
```

Comment:
Can we make good on this comment?

Reply:
I see how this is a difficult problem to solve. But I still really don't feel right about this work around